src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+
+    @BeforeAll
+    static void createReference() {
+        registrationService = new RegistrationServiceImpl();
+        userIdCounter = new AtomicLong(1);
+    }
+
+    @BeforeEach
+    void resetStorage() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_userAlreadyExists_shouldThrowException() {
+        User user = new User();
+        user.setId(userIdCounter.getAndIncrement());
+        user.setLogin("validUser123");
+        user.setPassword("password123");
+        user.setAge(20);
+        Storage.people.add(user);
+
+        User duplicateUser = new User();
+        duplicateUser.setLogin("validUser123");
+        duplicateUser.setPassword("password123");
+        duplicateUser.setAge(20);
+
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(duplicateUser));
+    }
+
+    @Test
+    void register_ageAboveMin_shouldAddUserToStorage() throws RegistrationException {
+        User user = new User();
+        user.setId(userIdCounter.getAndIncrement());
+        user.setLogin("validUser123");
+        user.setPassword("validPassword123");
+        user.setAge(20);
+
+        registrationService.register(user);
+
+        assertEquals(1, Storage.people.size());
+        assertEquals("validUser123", Storage.people.get(0).getLogin());
+    }
+
+    @Test
+    void register_passwordLengthAtMin_shouldSucceed() throws RegistrationException {
+        User user = new User();
+        user.setId(userIdCounter.getAndIncrement());
+        user.setLogin("userWithPass");
+        user.setPassword("pass12");
+        user.setAge(20);
+
+        registrationService.register(user);
+
+        assertEquals(1, Storage.people.size());
+        assertEquals("userWithPass", Storage.people.get(0).getLogin());
+    }
+
+    @Test
+    void register_passwordLengthBelowMin_shouldThrowException() {
+        User user = new User();
+        user.setLogin("userWithShortPass");
+        user.setPassword("pass");
+        user.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullLogin_shouldThrowException() {
+        User user = new User();
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullPassword_shouldThrowException() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageBelowMin_shouldThrowException() {
+        User user = new User();
+        user.setLogin("underageUser");
+        user.setPassword("validPassword");
+        user.setAge(15);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
```

Comment:
You are testing for an age below the minimum, which is good. However, make sure to also test the boundary conditions, such as an age exactly equal to the minimum and just above it, to cover all possible edge cases.